test/api/model/submessage_test.dart

```diff
@@ -111,4 +116,176 @@ void main() {
+      check(messageWithSubmessageContents(
+        pollWidgetData,
+        events: [
+          (eg.otherUser.userId, PollQuestionEventSubmessage(question: 'new question')),
+        ],
+      ))
+        .poll.isNotNull()
+          ..question.equals('new question')
+          ..options.deepEquals(defaultOptions);
+    });
+
+    test('parse poll with new option event', () {
+      check(messageWithSubmessageContents(
+        pollWidgetData,
+        events: [
+          (eg.otherUser.userId, PollNewOptionEventSubmessage(idx: 3, option: 'D')),
+          (eg.selfUser.userId, PollNewOptionEventSubmessage(idx: 0, option: 'E')),
+        ],
+      ))
+        .poll.isNotNull()
+          ..question.equals('favorite letter')
+          ..options.deepEquals([
+            ...defaultOptions,
+            PollOption(text: 'D'),
+            PollOption(text: 'E'),
+          ]);
+    });
+
+    test('parse poll with vote events on initial canned options', () {
+      check(messageWithSubmessageContents(
+        pollWidgetData,
+        events: [
+          (eg.otherUser.userId, PollVoteEventSubmessage(key: 'canned,1', op: PollVoteOp.add)),
+          (eg.otherUser.userId, PollVoteEventSubmessage(key: 'canned,2', op: PollVoteOp.add)),
+          (eg.otherUser.userId, PollVoteEventSubmessage(key: 'canned,2', op: PollVoteOp.remove)),
+          (eg.selfUser.userId, PollVoteEventSubmessage(key: 'canned,1', op: PollVoteOp.add)),
+        ],
+      ))
+        .poll.isNotNull()
+          ..question.equals('favorite letter')
+          ..options.deepEquals([
+            PollOption(text: 'A'),
+            eg.pollOption(text: 'B', voters: [eg.otherUser.userId, eg.selfUser.userId]),
+            PollOption(text: 'C'),
+          ]);
+    });
+
+    test('parse poll with vote events on post-creation options', () {
+      check(messageWithSubmessageContents(
+        pollWidgetData,
+        events: [
+          (eg.otherUser.userId, PollNewOptionEventSubmessage(idx: 0, option: 'D')),
+          (eg.otherUser.userId, PollVoteEventSubmessage(key: '${eg.otherUser.userId},0', op: PollVoteOp.add)),
+          (eg.otherUser.userId, PollVoteEventSubmessage(key: '${eg.selfUser.userId},0', op: PollVoteOp.add)),
+          (eg.otherUser.userId, PollVoteEventSubmessage(key: '${eg.selfUser.userId},0', op: PollVoteOp.remove)),
+        ],
+      ))
+        .poll.isNotNull()
+          ..question.equals('favorite letter')
+          ..options.deepEquals([
+            ...defaultOptions,
+            eg.pollOption(text: 'D', voters: [eg.otherUser.userId]),
+          ]);
+    });
+
+    test('applyEvent: adding repeated option', () {
+      final poll = messageWithSubmessageContents(pollWidgetData).poll!;
+      check(poll.options).deepEquals(defaultOptions);
+      poll.applyEvent(eg.otherUser.userId, PollNewOptionEventSubmessage(
+        option: defaultOptions[0].text, idx: 0));
+      check(poll.options).deepEquals(defaultOptions);
```

Comment:
In either the name or a one-line comment within the test, say explicitly how the story ends: IIUC, the answer is that the event gets ignored.

Reply:
This would do it:
```suggestion
    test('applyEvent: option with duplicate text ignored', () {
      final poll = messageWithSubmessageContents(pollWidgetData).poll!;
      check(poll.options).deepEquals(defaultOptions);
      poll.applyEvent(eg.otherUser.userId, PollNewOptionEventSubmessage(
        option: defaultOptions[0].text, idx: 0));
      check(poll.options).deepEquals(defaultOptions);
```